src/uu/cut/src/cut.rs

```diff
@@ -337,6 +353,101 @@ fn cut_fields<R: Read>(reader: R, ranges: &[Range], opts: &FieldOptions) -> URes
     Ok(())
 }
 
+#[allow(clippy::cognitive_complexity)]
+fn cut_fields<R: Read>(reader: R, ranges: &[Range], opts: &FieldOptions) -> UResult<()> {
+    let newline_char = if opts.zero_terminated { b'\0' } else { b'\n' };
+    match opts.delimiter {
+        Delimiter::Whitespace => cut_fields_whitespace(
+            reader,
+            ranges,
+            opts.only_delimited,
+            newline_char,
+            match opts.out_delimiter {
```

Comment:
Nit: This could be `opts.out_delimiter.unwrap_or("\t")`

Reply:
This does not seem to work, since `opts` is passed as immutable and `unwrap_or` consumes the option.

```
error[E0507]: cannot move out of `opts.out_delimiter` which is behind a shared reference
   --> src/uu/cut/src/cut.rs:365:14
    |
365 |             &opts.out_delimiter.unwrap_or("\t".to_owned())
    |              ^^^^^^^^^^^^^^^^^^ move occurs because `opts.out_delimiter` has type `Option<std::string::String>`, which does not implement the `Copy` trait

For more information about this error, try `rustc --explain E0507`.
error: could not compile `uu_cut` due to previous error
```

I could do clone:
```
&opts.out_delimiter.clone().unwrap_or("\t".to_owned())
```

or pass`opts` as mutable, but that would require more edits from the caller side as well

Reply:
There's no mutability involved here. It's just that Rust thinks you need ownershio because you ask for a `String`. `opts.out_delimiter.as_deref().unwrap_or("\t")` fixes that.

Reply:
oh I see. Done